### PR TITLE
SRE 000 do not return error if CR is removed

### PR DESCRIPTION
**Pull request by activehigh**
### Background ###

Check if VirtualService Merge exists before try to update status

Fixes [(issue)]

### What Has Changed: ###

List out what has changed as a result of this PR.

### How Has This Been Tested? ###

Describe the tests that you ran to verify your changes. 

### Notes

Add any notes or extra information here that might be useful to the reviewer if applicable i.e. links to documentation/blogs or dashboards.

### Checklist: ###

- [ ] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have verified that CI completes successfully.
- [ ] All insignificant commits have been squashed.

### CODE REVIEW

**Review by benpaternostro — APPROVED** [review]

**Comment by benpaternostro** [comment]
tests?

File: controller/controller.go [hunk 1]
```diff
@@ -137,12 +137,22 @@ func (r *VirtualServicePatchReconciler) Reconcile(ctx context.Context, request r
 		//trigger event
 		r.EventRecorder.Event(patch, "Warning", "ReconciliationFailed", fmt.Sprintf("VirtualServiceMerge reconcile error: %s", err.Error()))
 		if err2 := r.Context.Client().Get(ctx, request.NamespacedName, patch); err2 != nil {
+			if kerr.IsNotFound(err2) {
+				// do not need to panic just log output
+				r.Context.Logger().Info("Virtual service merge not found. No status to update.")
```
Comment (JonCubed) [inline controller/controller.go:142]: should we add details of what was now found?

Comment (activehigh) [reply controller/controller.go:142]: don't think so, it is kind of standard to always check for `not found error ` when writing controller codes. It was a mistake on my side that I forgot to add the check.

**Comment by activehigh** [comment]
> tests?

Unfortunately, the person who wrote the original code didn't use interface. So unit testing this part is kind of impossible.

But since you mentioned, I will give a try.

**Review by alexnguyenatwork — APPROVED** [review]

File: controller/controller.go (hunk 1 above)
Comment (alexnguyenatwork) [reply controller/controller.go:142]: no harm in printing out the which virtual service is involved when this not found error happens though ,right? +1

**Pull request merged**